js/packages/binderhub-client/lib/index.js

```diff
@@ -1,9 +1,21 @@
-import { NativeEventSource, EventSourcePolyfill } from "event-source-polyfill";
+import { fetchEventSource } from "@microsoft/fetch-event-source";
 import { EventIterator } from "event-iterator";
 
-// Use native browser EventSource if available, and use the polyfill if not available
-const EventSource = NativeEventSource || EventSourcePolyfill;
-
+function _getXSRFToken() {
+  // from @jupyterlab/services
+  let cookie = "";
+  try {
+    cookie = document.cookie;
+  } catch (e) {
+    // e.g. SecurityError in case of CSP Sandbox
+    return null;
+  }
+  const xsrfTokenMatch = cookie.match("\\b_xsrf=([^;]*)\\b");
```

Comment:
regexes in anything security sensitive always make me nervous haha. Can you either add a comment here explaining what's going on, or a direct link to the specific place from @jupyterlab/services this was stolen from?

Reply:
added link and explanation. It's still astonishing to me that there is no API for geting cookies by name in browsers, but it is `{name}={anything but a semicolon}[; again]`